x-pack/plugins/canvas/public/services/storybook/workpad.ts

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import moment from 'moment';
+import { PluginServiceFactory } from '../../../../../../src/plugins/presentation_util/public';
+
+import { StorybookParams } from '.';
+import { CanvasWorkpadService } from '../workpad';
+// @ts-expect-error
+import { getDefaultWorkpad } from '../../state/defaults';
+import { getId } from '../../lib/get_id';
+import { CanvasTemplate } from '../../../types';
+
+type CanvasWorkpadServiceFactory = PluginServiceFactory<CanvasWorkpadService, StorybookParams>;
+
+const TIMEOUT = 500;
+
+const promiseTimeout = (time: number) => () => new Promise((resolve) => setTimeout(resolve, time));
+const getName = () => {
+  const lorem = 'Lorem ipsum dolor sit amet consectetur adipiscing elit Fusce lobortis aliquet arcu ut turpis duis'.split(
+    ' '
+  );
+  return [1, 2, 3].map(() => lorem[Math.floor(Math.random() * lorem.length)]).join(' ');
+};
+
```

Comment:
nit/future: this is clever, but it kills storyshots.  Add arg to swap between random and established data?